pkg/labels/labels.go

```diff
@@ -292,6 +292,13 @@ func NewBuilder(base Labels) *Builder {
 	}
 }
 
+// Reset clears all current state for the builder
+func (b *Builder) Reset(base Labels) {
+	b.base = base
+	b.del = b.del[:]
```

Comment:
Did you mean to write `b.del[:0]`? Same for `b.add`.

Reply:
This is actually intentional, this creates an empty slice referencing the same storage as the previous slice (some details in https://blog.golang.org/go-slices-usage-and-internals).

Reply:
I think `[:0]` does what you said. `[:]` doesn't give empty slice if the original slice was not empty.
https://play.golang.org/p/seYfZEZczpi

Reply:
Ah, you are correct. Interesting that none of the tests caught this though ;) I have updated the PR